Adaptive Card Designer - Host App for Federated Search Channel

naramaka opened the pull request:
**Creating host app for Federated Search Channel:**
The change is for new requirement to include preview card for Federate Search Channel in adaptive card designer app.
Added new card in designer template for Federated Search Channel.
Developers can use this to preview the adaptive card response on Search channel before actually deploying to prod.

Verified manually the layout . Attaching the screen shot as per current configuration.
![Federated-Search-Card-Preview](https://user-images.githubusercontent.com/65971640/155198016-65b7ecd2-89ab-40ef-9e52-d33d7baca432.jpg)
![Federated-Search-Channel](https://user-images.githubusercontent.com/65971640/155198314-59ef0b06-5b82-4516-a5aa-441722fbe0cb.jpg)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/7061)

dclaux commented on source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:
```diff
@@ -0,0 +1,140 @@
+  }
+
+  public renderTo(hostElement: HTMLElement) {
+    hostElement.classList.add("outlook-frame");
```
Why use an "outlook" branded frame when this HostContainer is for Federated Search? Plus this class doesn'r even seem to be defined in your css.

naramaka replied:
Removed outlook related code.

dclaux commented on source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:
```diff
@@ -0,0 +1,140 @@
+    Adaptive.GlobalSettings.useMarkdownInRadioButtonAndCheckbox = false;
+  }
+
+  private parsePadding(source: any): Adaptive.PaddingDefinition {
```
This seems to have been copied straight from the Outlook host container. I doubt Federated Search cards support explicit padding (and if they do they probably shouldn't). This method likely should be removed.

dclaux commented on source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:
```diff
@@ -0,0 +1,140 @@
+    source: any,
+    context: Adaptive.SerializationContext
+  ) {
+    if (element instanceof Adaptive.Container && typeof source["rtl"] === "boolean") {
```
This isn't necessary. The `rtl` property has been an official AC property for a while and is parsed automatically.

naramaka replied:
Removed.

dclaux commented on source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:
```diff
@@ -0,0 +1,140 @@
+      element.rtl = source["rtl"];
+    }
+
+    if (element instanceof Adaptive.AdaptiveCard) {
```
All this needs to be removed... It's something that only Outlook supports.

naramaka replied:
Removed.

dclaux reviewed (changes requested):
Beyond my comments, the changes modify the formatting of certain existing files. Please revert to the previous formatting.

naramaka commented:
Hi David ,

I had no instructions related to Federated Search changes so I had copied directly from Outlook. I have resolved the review comments.

But to your point to revert to the previous the formatting , for example I reverted the changes did to app.ts  and removed // eslint-disable-next-line no-console.
And after that it is not allowing to commit hence I had to make the changes I have made to existing files. Please find the error below:

> git -c user.useConfigOnly=true commit --quiet --allow-empty-message --file -
[STARTED] Preparing...
[SUCCESS] Preparing...
[STARTED] Running tasks...
[STARTED] Running tasks for **/!(package-lock)*
[STARTED] eslint --fix --max-warnings=0 C:/MyProjects/AdaptivecardRepo/AdaptiveCard…
[FAILED] [91meslint --fix --max-warnings=0 C:/MyProjects/AdaptivecardRepo/AdaptiveCards/source/nodejs/adaptivecards-designer-app/src/app.ts C:/MyProjects/AdaptivecardRepo/AdaptiveCards/source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts[39m [2m[FAILED][22m
[SUCCESS] Running tasks...
[STARTED] Applying modifications...
[SKIPPED] Skipped because of errors from tasks.
[STARTED] Reverting to original state because of errors...
[SUCCESS] Reverting to original state because of errors...
[STARTED] Cleaning up...
[SUCCESS] Cleaning up...

[91m[91m×[91m eslint --fix --max-warnings=0 C:/MyProjects/AdaptivecardRepo/AdaptiveCards/source/nodejs/adaptivecards-designer-app/src/app.ts C:/MyProjects/AdaptivecardRepo/AdaptiveCards/source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:[39m

C:\MyProjects\AdaptivecardRepo\AdaptiveCards\source\nodejs\adaptivecards-designer-app\src\app.ts
  52:5  error  Unexpected console statement  no-console

✖ 1 problem (1 error, 0 warnings)

husky - pre-commit hook exited with code 1 (error)

Kindly let me know how can I resolve this.

Best Regards,
Nayana.
From: David Claux ***@***.***>
Sent: Wednesday, February 23, 2022 10:39 PM
To: microsoft/AdaptiveCards ***@***.***>
Cc: Nayana Ramakanth ***@***.***>; Mention ***@***.***>
Subject: Re: [microsoft/AdaptiveCards] Adaptive Card Designer - Host App for Federated Search Channel (PR #7061)


@dclaux requested changes on this pull request.

Beyond my comments, the changes modify the formatting of certain existing files. Please revert to the previous formatting.

________________________________

In source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FAdaptiveCards%2Fpull%2F7061%23discussion_r813111687&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270565677%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=9Cd7ZSN7kWq8%2F8disJy1MNAOWV5BvNJ%2B0SjAIbvn%2BRA%3D&reserved=0>:

> +import * as Adaptive from "adaptivecards";

+import { HostContainer } from "../host-container";

+import * as hostConfig from "../../hostConfigs/federated-search.json";

+

+export class FederatedSearchContainer extends HostContainer {

+  constructor(name: string, styleSheet: string) {

+    super(name, styleSheet);

+

+    this.actionsRegistry.unregister("Action.Submit");

+    this.elementsRegistry.unregister("Input.Text");

+    this.elementsRegistry.unregister("Input.Date");

+    this.elementsRegistry.unregister("Input.Time");

+    this.elementsRegistry.unregister("Input.Toggle");

+    this.elementsRegistry.unregister("Input.ChoiceSet");

+    this.elementsRegistry.unregister("Input.Number");

+    this.actionsRegistry.register("Action.Http", Adaptive.HttpAction);

Why would Federated search support Action.Http? Shouldn't FS only support Action.Execute? Action.Http was used solely by Outlook and we have now moved to the universal action model (Action.Execute) which is what FS should use.

________________________________

In source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FAdaptiveCards%2Fpull%2F7061%23discussion_r813112778&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270565677%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=3raKPA224PtxdpEsItmeAg0FS52aJd1guG%2BNOR%2B%2FAMU%3D&reserved=0>:

> +  constructor(name: string, styleSheet: string) {

+    super(name, styleSheet);

+

+    this.actionsRegistry.unregister("Action.Submit");

+    this.elementsRegistry.unregister("Input.Text");

+    this.elementsRegistry.unregister("Input.Date");

+    this.elementsRegistry.unregister("Input.Time");

+    this.elementsRegistry.unregister("Input.Toggle");

+    this.elementsRegistry.unregister("Input.ChoiceSet");

+    this.elementsRegistry.unregister("Input.Number");

+    this.actionsRegistry.register("Action.Http", Adaptive.HttpAction);

+    this.cardHost.classList.add("fedsearch-card");

+  }

+

+  public renderTo(hostElement: HTMLElement) {

+    hostElement.classList.add("outlook-frame");

Why use an "outlook" branded frame when this HostContainer is for Federated Search? Plus this class doesn'r even seem to be defined in your css.

________________________________

In source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FAdaptiveCards%2Fpull%2F7061%23discussion_r813114367&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270565677%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=d7RNb5lQnMoDTpFU9DqhDNewSvSAUxvZt6H2I%2BRZ4lc%3D&reserved=0>:

> +    this.actionsRegistry.register("Action.Http", Adaptive.HttpAction);

+    this.cardHost.classList.add("fedsearch-card");

+  }

+

+  public renderTo(hostElement: HTMLElement) {

+    hostElement.classList.add("outlook-frame");

+    hostElement.appendChild(this.cardHost);

+  }

+

+  public initialize() {

+    super.initialize();

+

+    Adaptive.GlobalSettings.useMarkdownInRadioButtonAndCheckbox = false;

+  }

+

+  private parsePadding(source: any): Adaptive.PaddingDefinition {

This seems to have been copied straight from the Outlook host container. I doubt Federated Search cards support explicit padding (and if they do they probably shouldn't). This method likely should be removed.

________________________________

In source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FAdaptiveCards%2Fpull%2F7061%23discussion_r813115459&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270565677%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=Rx57xOUEdhN4zy%2BlZhArFeZeO6HETmDNQ%2FKXyhPPO8U%3D&reserved=0>:

> +          Adaptive.parseEnum(Adaptive.Spacing, source["right"], Adaptive.Spacing.None),

+          Adaptive.parseEnum(Adaptive.Spacing, source["bottom"], Adaptive.Spacing.None),

+          Adaptive.parseEnum(Adaptive.Spacing, source["left"], Adaptive.Spacing.None)

+        );

+      }

+    }

+

+    return null;

+  }

+

+  public parseElement(

+    element: Adaptive.CardElement,

+    source: any,

+    context: Adaptive.SerializationContext

+  ) {

+    if (element instanceof Adaptive.Container && typeof source["rtl"] === "boolean") {

This isn't necessary. The rtl property has been an official AC property for a while and is parsed automatically.

________________________________

In source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FAdaptiveCards%2Fpull%2F7061%23discussion_r813116193&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270565677%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=w15xFjfWh5ydyulCnlpQJIqB4nc3ZSrFUKxzjUI9VxA%3D&reserved=0>:

> +      }

+    }

+

+    return null;

+  }

+

+  public parseElement(

+    element: Adaptive.CardElement,

+    source: any,

+    context: Adaptive.SerializationContext

+  ) {

+    if (element instanceof Adaptive.Container && typeof source["rtl"] === "boolean") {

+      element.rtl = source["rtl"];

+    }

+

+    if (element instanceof Adaptive.AdaptiveCard) {

All this needs to be removed... It's something that only Outlook supports.

________________________________

In source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FAdaptiveCards%2Fpull%2F7061%23discussion_r813116356&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270565677%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=GP4egeYq%2Bi7XJ2nn01IxtTkYxhYpkhSSB0BeFLvVFbg%3D&reserved=0>:

> +        const actionResources = source["resources"]["actions"] as Array<any>;

+

+        actionResources.forEach((action) => {

+          this.actionsRegistry.createInstance(actionResources["type"], context.targetVersion);

+

+          if (action) {

+            action.parse(actionResources, context);

+            action.setParent(card);

+

+            actionArray.push(action);

+          }

+        });

+      }

+    }

+

+    if (element instanceof Adaptive.Image) {

Remove

________________________________

In source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FAdaptiveCards%2Fpull%2F7061%23discussion_r813116476&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270565677%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=AEy3YS9hLEKwmJVjeInRoIer6DlVZb9xUCBEwIB226Y%3D&reserved=0>:

> +

+          if (action) {

+            action.parse(actionResources, context);

+            action.setParent(card);

+

+            actionArray.push(action);

+          }

+        });

+      }

+    }

+

+    if (element instanceof Adaptive.Image) {

+      element.backgroundColor = source["backgroundColor"];

+    }

+

+    if (element instanceof Adaptive.Container) {

Remove

________________________________

In source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FAdaptiveCards%2Fpull%2F7061%23discussion_r813116604&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270565677%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=sbVIyf3USm2ugkw7VEAMwAxQ%2BXUImGszlR3gIZkw3zI%3D&reserved=0>:

> +      }

+    }

+

+    if (element instanceof Adaptive.Image) {

+      element.backgroundColor = source["backgroundColor"];

+    }

+

+    if (element instanceof Adaptive.Container) {

+      const padding = this.parsePadding(source["padding"]);

+

+      if (padding) {

+        element.padding = padding;

+      }

+    }

+

+    if (element instanceof Adaptive.ColumnSet) {

Remove

________________________________

In source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FAdaptiveCards%2Fpull%2F7061%23discussion_r813116964&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270615664%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=1xx3kCXVGx1X4l3EIzwqPwtsWZJkPtmdzfizNqy7Dh8%3D&reserved=0>:

> +

+      if (padding) {

+        element.padding = padding;

+      }

+    }

+

+    if (element instanceof Adaptive.ColumnSet) {

+      const padding = this.parsePadding(source["padding"]);

+

+      if (padding) {

+        element.padding = padding;

+      }

+    }

+  }

+

+  public anchorClicked(element: Adaptive.CardElement, anchor: HTMLAnchorElement): boolean {

Remove

—
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FAdaptiveCards%2Fpull%2F7061%23pullrequestreview-891388848&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270615664%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=BIJIrZoFhZVJ%2FKf0mxSm3InrJ1INrhaRX%2BPSW0xTXG8%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAPXKLOBOHW26SOCFANDCLVDU4UIBVANCNFSM5PEDJFXQ&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270615664%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=Qrs3sbIfCThHhjf97%2B%2FHWGDPBqwnZcUND01niuUg7Kk%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270615664%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=zfZm073vR7EA4m2K5xJIqoDpQfYKJSkOS7XJhFuBXGI%3D&reserved=0> or Android<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7CNayana.Ramakanth%40microsoft.com%7Cfb8e9b1e41a24182680308d9f6ef24b6%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637812329270615664%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=nbGYJbfjxdaRsbqtC%2FgsfEv5HQu%2BgZpAfql68RgImr8%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

dclaux commented:
@naramaka sorry, but I have no idea how reverting to the original formatting could possibly prevent the code from building. I'm not going to be able to help with that.

Thanks for making the code changes. However, without a new commit, I can't review your last changes.

naramaka commented on source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:
```diff
@@ -0,0 +1,140 @@
+    this.elementsRegistry.unregister("Input.Toggle");
+    this.elementsRegistry.unregister("Input.ChoiceSet");
+    this.elementsRegistry.unregister("Input.Number");
+    this.actionsRegistry.register("Action.Http", Adaptive.HttpAction);
```
Its not needed. Removed it.

naramaka commented on source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:
```diff
@@ -0,0 +1,140 @@
+      }
+    }
+
+    if (element instanceof Adaptive.Image) {
```
Removed

naramaka commented on source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:
```diff
@@ -0,0 +1,140 @@
+      element.backgroundColor = source["backgroundColor"];
+    }
+
+    if (element instanceof Adaptive.Container) {
```
Removed

naramaka commented on source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:
```diff
@@ -0,0 +1,140 @@
+    }
+  }
+
+  public anchorClicked(element: Adaptive.CardElement, anchor: HTMLAnchorElement): boolean {
```
Removed.

naramaka commented on source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:
```diff
@@ -0,0 +1,140 @@
+      }
+    }
+
+    if (element instanceof Adaptive.ColumnSet) {
```
Removed

dclaux commented:
@naramaka sorry, I had missed it (don't know how...)

dclaux commented:
@naramaka just to be suere - this is the styling you're going for in search for buttons?
![image](https://user-images.githubusercontent.com/1334689/156039639-094f96e2-0952-4f1a-b8d9-2fd5649936e2.png)

Asking because they don't necessarily look like buttons I tend to see in other places, like in Office.com for instance:
![image](https://user-images.githubusercontent.com/1334689/156039882-d9af34ef-566a-4f37-a345-eb64346baff7.png)

Also, the font used in your buttons isn't the same as the font used in the rest of the card.

naramaka commented:
> @naramaka just to be suere - this is the styling you're going for in search for buttons? ![image](https://user-images.githubusercontent.com/1334689/156039639-094f96e2-0952-4f1a-b8d9-2fd5649936e2.png)
> 
> Asking because they don't necessarily look like buttons I tend to see in other places, like in Office.com for instance: ![image](https://user-images.githubusercontent.com/1334689/156039882-d9af34ef-566a-4f37-a345-eb64346baff7.png)
> 
> Also, the font used in your buttons isn't the same as the font used in the rest of the card.

@dclaux There is no specification yet. I am submitting initial configuration so that we can improve it as and when we have the clarity. But if you do have any default specification then let me know I can change.

paulcam206 commented:
@licanhua - can you review?
